wcfsetup/install/files/lib/system/cache/eager/data/LanguageCacheData.class.php

```diff
@@ -62,6 +60,68 @@ public function getLanguageCategory(string $categoryName): ?LanguageCategory
      */
     public function getLanguageCategoryByID(int $languageCategoryID): ?LanguageCategory
     {
-        return $this->categories[$this->categoryIDs[$languageCategoryID] ?? null] ?? null;
+        $categoryName = $this->categoryIDs[$languageCategoryID] ?? null;
+        if ($categoryName === null) {
+            return null;
+        }
+
+        return $this->getLanguageCategory($categoryName);
+    }
+
+    /**
+     * Returns `true` if the language category with the given category name exists.
+     */
+    public function languageCategoryExists(string $categoryName): bool
+    {
+        return \array_key_exists($categoryName, $this->categories);
+    }
+
+    /**
+     * Return all content languages.
+     *
+     * @return list<Language>
```

Comment:
```suggestion
     * @return array<int, Language>
```

The `list<…>` is incorrect for two reasons:

1. `$this->languages` is actually a mapping of the language id to the object.
2. `\array_filter()` preserves the keys so even if `$this->languages` was a zero-based array without holes, it would not longer be guaranteed afterwards.

`list<…>` indicates that the array only has numeric indices, starts at zero and has no holes in it. This is meant to provide type safety when iterating over it using a simple `for` loop.